### PR TITLE
CMake: Add "test" target

### PR DESCRIPTION
**Pull request by ctrlaltca**
Adds a new "test" target that can be enabled from cmake ``` -DWITH_TESTS=1```.
CMake downloads a vendored copy of googletest if not found on the system.
Tests can be run using ```make test```.
Also, update qt from 5.2 to 5.4.1 (from the same ppa)
Replaces and fixes #1600.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -232,3 +232,10 @@ if(WITH_ORACLE)
     add_subdirectory(oracle)
     SET(CPACK_INSTALL_CMAKE_PROJECTS "release/oracle.app;oracle;ALL;/" ${CPACK_INSTALL_CMAKE_PROJECTS})
 endif()
+
+# Compile tests (default off)
+option(WITH_TESTS "build tests" OFF)
+if(WITH_TESTS)
```
Comment (Daenyth) [inline CMakeLists.txt]: Why use this over the built in `-Dtest=ON`?

Comment (ctrlaltca) [inline CMakeLists.txt]: Afaik no special "test" variable exists in cmake: https://cmake.org/cmake/help/v3.0/manual/cmake-variables.7.html
A special "test" target exists when CTest is enabled, but "-Dtest=ON" is nothing more than a normal #define.
So, tried to make it homogeneous with the other target-enabling definitions  (WITH_ORACLE, WITH_SERVER, ..).

Comment (Daenyth) [inline CMakeLists.txt]: I don't have a strong feeling about it.

Comment (ctrlaltca) [inline CMakeLists.txt]: Reverted to "TEST"

File: travis-compile.sh [hunk 2]
```diff
@@ -9,13 +9,14 @@ if [[ $TRAVIS_OS_NAME == "osx" && $QT4 == 0 ]]; then
   prefix="-DCMAKE_PREFIX_PATH=`echo /usr/local/Cellar/qt5/5.*/`"
 fi
 if [[ $TRAVIS_OS_NAME == "linux" && $QT4 == 0 ]]; then
-  prefix="-DCMAKE_PREFIX_PATH=/opt/qt52/lib/cmake/"
+  prefix="-DCMAKE_PREFIX_PATH=`echo /opt/qt5*/lib/cmake/`"
 fi
-cmake .. -DWITH_SERVER=1 -DCMAKE_BUILD_TYPE=$BUILDTYPE -DWITH_QT4=$QT4 $prefix
 
 if [[ $BUILDTYPE == "Debug" ]]; then
```
Comment (Daenyth) [inline travis-compile.sh]: :+1: I like the split.

**Comment by Daenyth** [comment]
Can you update the README with the new cmake option and then :shipit:

**Pull request merged**